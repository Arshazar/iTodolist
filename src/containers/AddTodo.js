import React, { useState } from 'react'
import { connect } from 'react-redux'
import {
    Button,
    Modal,
    ModalFooter,
    ModalHeader,
    ModalBody,
    Input,
    Label,
    Form,
    FormGroup,
} from 'reactstrap'
import { addTodo } from '../actions'

const AddTodo = ({ dispatch, modal, toggle }) => {
    let [todoText, changeText] = useState('')

    const onTextChange = event => {
        changeText(event.target.value)
    }

    const onSubmit = e => {
        e.preventDefault()
        if (!todoText.trim()) {
            return
        }
        dispatch(addTodo(todoText))
        changeText('')
        toggle(false)
    }

    return (
        <div>
            <Modal
                isOpen={modal}
                toggle={toggle}
                centered={true}
                className="h-50 col-lg-4 col-md-6 col-sm-10"
                fade={true}
            >
                <ModalHeader>Add todo</ModalHeader>
                <Form>
                    <ModalBody>
                        <FormGroup>
                            <Label for="writing">Write your todo</Label>
                            <Input
                                style={{ boxSizing: 'border-box' }}
                                type="textarea"
                                id="todo-text"
                                onChange={onTextChange}
                            />
                        </FormGroup>
                    </ModalBody>
                    <ModalFooter className="justify-content-between">
                        <Button color="primary" onClick={onSubmit}>
                            Add
                        </Button>
                        <Button className="ml-5" onClick={toggle}>
                            Cancel
                        </Button>
                    </ModalFooter>
                </Form>
            </Modal>
        </div>
    )
}

export default connect()(AddTodo)
