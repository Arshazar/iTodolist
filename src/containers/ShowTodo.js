import React, { useState } from 'react'
import { connect } from 'react-redux'
import {
    Button,
    Modal,
    ModalFooter,
    ModalHeader,
    ModalBody,
    Input,
    Form,
    FormGroup,
} from 'reactstrap'
import { editTodo, removeTodo, toggleTodo } from '../actions'

const ShowTodo = ({ dispatch, completed, modal, toggle, id, text }) => {
    let [todoText, changeText] = useState(text)

    const onToggleTodo = () => {
        dispatch(toggleTodo(id))
        toggle(false)
    }

    const onTextChange = event => {
        changeText(event.target.value)
    }

    const onEditTodo = () => {
        if (!todoText.trim()) {
            return
        }
        dispatch(editTodo(id, todoText))
        toggle(false)
    }

    const onRemoveTodo = () => {
        dispatch(removeTodo(id))
        toggle(false)
    }

    return (
        <div>
            <Modal
                isOpen={modal}
                toggle={toggle}
                centered={true}
                className="h-50 col-lg-4 col-md-6 col-sm-10"
                fade={true}
            >
                <ModalHeader>Your todo</ModalHeader>
                <Form>
                    <ModalBody>
                        <FormGroup>
                            <Input
                                style={{ boxSizing: 'border-box' }}
                                type="textarea"
                                id="todo-text"
                                placeholder={text}
                                onChange={onTextChange}
                            />
                        </FormGroup>
                    </ModalBody>
                    <ModalFooter className="justify-content-between">
                        <Button color="primary" onClick={onEditTodo}>
                            Save
                        </Button>
                        <Button color="danger" onClick={onRemoveTodo}>
                            Remove
                        </Button>
                        {!completed ? (
                            <Button color="success" onClick={onToggleTodo}>
                                Complete
                            </Button>
                        ) : (
                            <Button color="warning" onClick={onToggleTodo}>
                                Incomplete
                            </Button>
                        )}
                        <Button onClick={toggle}>Cancel</Button>
                    </ModalFooter>
                </Form>
            </Modal>
        </div>
    )
}

export default connect()(ShowTodo)
