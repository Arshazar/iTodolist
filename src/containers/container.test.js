import React from 'react'
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'

import AddTodo from './AddTodo'
import Link from './Link'
import ShowTodo from './ShowTodo'
import VisibleTodoList from './VisibleTodoList'

const addTodo = () => ({ type: 'ADD_TODO' })
const toggleTodo = () => ({ type: 'TOGGLE_TODO' })
const removeTodo = () => ({ type: 'REMOVE_TODO' })
const mockStore = configureStore()

let wrapper

it('renders AddTodo', () => {
    const mockProps = {
        addTodo: jest.fn(),
    }
    let store = mockStore(mockProps)
    store.dispatch(addTodo())
    wrapper = shallow(<AddTodo store={store} />)

    expect(wrapper).toMatchSnapshot()
})

it('renders ShowTodo component', () => {
    const mockProps = {
        toggleTodo: jest.fn(),
        removeTodo: jest.fn(),
    }
    let store = mockStore(mockProps)
    store.dispatch(toggleTodo())
    store.dispatch(removeTodo())
    wrapper = shallow(<ShowTodo store={store} />)

    expect(wrapper).toMatchSnapshot()
})

it('renders Undo component', () => {
    const mockProps = {
        onUndo: jest.fn(),
    }
    let store = mockStore(mockProps)
    wrapper = shallow(<ShowTodo store={store} />)

    expect(wrapper).toMatchSnapshot()
})

it('renders VisibleTodoList component', () => {
    let todos = [
        {
            text: 'first',
            completed: true,
        },
        {
            text: 'two',
            completed: false,
        },
    ]
    const mockProps = {
        todos: todos,
        visibilityFilter: ['SHOW_ALL', 'SHOW_COMPLETED', 'SHOW_ACTIVE'],
    }
    let store = mockStore(mockProps)
    wrapper = shallow(<VisibleTodoList store={store} />)

    expect(wrapper).toMatchSnapshot()
})
