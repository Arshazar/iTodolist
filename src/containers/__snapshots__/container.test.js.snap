// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AddTodo 1`] = `ShallowWrapper {}`;

exports[`renders ShowTodo component 1`] = `ShallowWrapper {}`;

exports[`renders Undo component 1`] = `ShallowWrapper {}`;

exports[`renders VisibleTodoList component 1`] = `ShallowWrapper {}`;
